internal/memorylimiter/memorylimiter_test.go

```diff
@@ -0,0 +1,188 @@
+
+	// Restore ballast to default.
+	ml.ballastSize = 0
+
+	// Check spike limit
+	ml.usageChecker.memSpikeLimit = 512
+
+	// Below memSpikeLimit.
+	currentMemAlloc = 500
+	ml.checkMemLimits()
+	assert.False(t, ml.MustRefuse())
+
+	// Above memSpikeLimit.
+	currentMemAlloc = 550
+	ml.checkMemLimits()
+	assert.True(t, ml.MustRefuse())
+}
+
+func TestGetDecision(t *testing.T) {
+	t.Run("fixed_limit", func(t *testing.T) {
+		d, err := getMemUsageChecker(&Config{MemoryLimitMiB: 100, MemorySpikeLimitMiB: 20}, zap.NewNop())
+		require.NoError(t, err)
+		assert.Equal(t, &memUsageChecker{
+			memAllocLimit: 100 * mibBytes,
+			memSpikeLimit: 20 * mibBytes,
+		}, d)
+	})
+
+	t.Cleanup(func() {
+		GetMemoryFn = iruntime.TotalMemory
+	})
+	GetMemoryFn = func() (uint64, error) {
+		return 100 * mibBytes, nil
+	}
+	t.Run("percentage_limit", func(t *testing.T) {
+		d, err := getMemUsageChecker(&Config{MemoryLimitPercentage: 50, MemorySpikePercentage: 10}, zap.NewNop())
+		require.NoError(t, err)
+		assert.Equal(t, &memUsageChecker{
+			memAllocLimit: 50 * mibBytes,
+			memSpikeLimit: 10 * mibBytes,
+		}, d)
+	})
+}
+
+func TestRefuseDecision(t *testing.T) {
+	decison1000Limit30Spike30 := newPercentageMemUsageChecker(1000, 60, 30)
+	decison1000Limit60Spike50 := newPercentageMemUsageChecker(1000, 60, 50)
+	decison1000Limit40Spike20 := newPercentageMemUsageChecker(1000, 40, 20)
+
+	tests := []struct {
+		name         string
+		usageChecker memUsageChecker
+		ms           *runtime.MemStats
+		shouldRefuse bool
+	}{
+		{
+			name:         "should refuse over limit",
+			usageChecker: *decison1000Limit30Spike30,
+			ms:           &runtime.MemStats{Alloc: 600},
+			shouldRefuse: true,
+		},
+		{
+			name:         "should not refuse",
+			usageChecker: *decison1000Limit30Spike30,
+			ms:           &runtime.MemStats{Alloc: 100},
+			shouldRefuse: false,
+		},
+		{
+			name: "should not refuse spike, fixed usageChecker",
+			usageChecker: memUsageChecker{
+				memAllocLimit: 600,
+				memSpikeLimit: 500,
+			},
+			ms:           &runtime.MemStats{Alloc: 300},
+			shouldRefuse: true,
+		},
+		{
+			name:         "should refuse, spike, percentage usageChecker",
+			usageChecker: *decison1000Limit60Spike50,
+			ms:           &runtime.MemStats{Alloc: 300},
+			shouldRefuse: true,
+		},
+		{
+			name:         "should refuse, spike, percentage usageChecker",
+			usageChecker: *decison1000Limit40Spike20,
+			ms:           &runtime.MemStats{Alloc: 250},
+			shouldRefuse: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			shouldRefuse := test.usageChecker.aboveSoftLimit(test.ms)
+			assert.Equal(t, test.shouldRefuse, shouldRefuse)
+		})
+	}
+}
+
+func TestBallastSize(t *testing.T) {
+	cfg := &Config{
+		CheckInterval:  10 * time.Second,
+		MemoryLimitMiB: 1024,
+	}
+	got, err := NewMemoryLimiter(cfg, zap.NewNop())
+	require.NoError(t, err)
+
+	got.setBallastSizeIfExtensionExists(&host{ballastSize: 113})
+	got.startMonitoring()
```

Comment:
To match the original test
```suggestion
	require.NoError(t, err)
	require.NoError(t, got.Start(context.Background(), &host{ballastSize: 113}))
```

Reply:
Please add `assert.Equal(t, uint64(113), got.ballastSize)` as in the original test

Reply:
added